### PR TITLE
RELATED: ONE-3331 Refactor of coloring 

### PR DESCRIPTION
**Pull request by ivanmjartan**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/1566/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1050/ : SUCCESS in 1m 33s

**Comment by ivanmjartan** [comment]
extended test - cafe

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/1567/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/322/ : FAILURE in 19s

**Comment by ivan-nejezchleb** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/612/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/1571/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/324/ : SUCCESS in 14m 03s

File: src/components/visualizations/chart/test/colorFactory.spec.ts [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+    VIEW_BY_DIMENSION_INDEX,
+    STACK_BY_DIMENSION_INDEX
+} from '../constants';
+
+import {
+    findMeasureGroupInDimensions,
+    findAttributeInDimension
+} from '../chartOptionsBuilder';
+
+function getMVS(dataSet: any) {
```
Comment (ivan-nejezchleb) [inline src/components/visualizations/chart/test/colorFactory.spec.ts]: this is the same function like in chartOptionsBuilder.spec file. Move it to the helper and import in both test files

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/1572/ : SUCCESS in 13s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1056/ : SUCCESS in 10m 04s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/1573/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1057/ : SUCCESS in 2m 53s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/1574/ : SUCCESS in 16s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1058/ : SUCCESS in 2m 39s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/1575/ : SUCCESS in 40s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1059/ : SUCCESS in 4m 04s

**Comment by ivan-nejezchleb** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/613/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/1576/ : SUCCESS in 18s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/325/ : SUCCESS in 16m 10s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-opensource-repository-check/1585/ : SUCCESS in 7s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1066/ : SUCCESS in 2m 02s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/351/ : SUCCESS in 23s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/1073/ : SUCCESS in 2m 12s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/621/ : SUCCESS in 10m 46s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/265/ : SUCCESS in 10m 34s

**Pull request merged**